apps/cowswap-frontend/src/modules/swap/containers/TradeButtons/swapTradeButtonsMap.tsx

```diff
@@ -26,12 +26,34 @@ interface SwapTradeButtonsContext {
   hasEnoughWrappedBalanceForSwap: boolean
   onCurrencySelection: (field: Field, currency: Currency) => void
   confirmText: string
+  isSafeWallet: boolean
+  isCurrentTradeBridging: boolean
 }
 
 type SwapTradeButton = (props: SwapTradeButtonsContext, isDisabled: boolean) => ReactNode | string
 
 export const swapTradeButtonsMap: Record<SwapFormState, SwapTradeButton> = {
   [SwapFormState.SwapWithWrappedToken]: (props: SwapTradeButtonsContext, isDisabled: boolean) => {
+    // When using Safe wallet (e.g., Safe+Rabby or Safe mobile + WC), the flow will wrap first
+    // So we should show "Wrap WETH and Swap" or "Wrap WETH and Swap and Bridge" instead of "Swap with WETH"
+    if (props.isSafeWallet) {
+      return (
+        <ButtonError buttonSize={ButtonSize.BIG} onClick={props.onEthFlow} disabled={isDisabled}>
+          <div>
+            {props.isCurrentTradeBridging ? (
+              <Trans>
```

Comment:
have you tried to build it? I think there are some i18n updates

Reply:
Pushed a new commit with the changes of `apps/cowswap-frontend/src/locales/en-US.po`